dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.c

```diff
@@ -34,6 +41,14 @@ RecycleBin5File_RecycleBinFile_QueryInterface(
         *ppvObject = &s->recycleBinFileImpl;
     else if (IsEqualIID(riid, &IID_IRecycleBinFile))
         *ppvObject = &s->recycleBinFileImpl;
+    else if (IsEqualIID(riid, &IID_IExtractIconA) || IsEqualIID(riid, &IID_IExtractIconW))
+    {
+        DWORD dwAttributes;
+        if (RecycleBin5File_RecycleBinFile_GetAttributes(This, &dwAttributes) == S_OK)
+            return SHCreateFileExtractIconW(s->FullName, (dwAttributes & FILE_ATTRIBUTE_DIRECTORY) ? FILE_ATTRIBUTE_DIRECTORY : FILE_ATTRIBUTE_NORMAL, riid, ppvObject);
```

Comment:
```suggestion
            return SHCreateFileExtractIconW(s->FullName, dwAttributes, riid, ppvObject);
```

Reply:
Why?
That changes what the code is doing.

Reply:
It's ok, the attributes are the same, so no need to and and select otyer attributes, you can use the attributes directly

Reply:
So hidden / system files do not influence this?

Reply:
Nope, i mean, it's supposed that the recycled is hidden and it works this way

Reply:
so it's done